tests/unit/obuf-limits.tcl

```diff
@@ -61,24 +71,65 @@ start_server {tags {"obuf-limits"}} {
         set start_time 0
         set time_elapsed 0
         while 1 {
-            if {$start_time != 0} {
-                # Slow down loop when omen has reached the limit.
-                after 10
-            }
             r publish foo [string repeat "x" 1000]
             set clients [split [r client list] "\r\n"]
-            set c [split [lindex $clients 1] " "]
+            set c [lsearch -inline $clients *name=test_client*]
             if {![regexp {omem=([0-9]+)} $c - omem]} break
             if {$omem > 100000} {
                 if {$start_time == 0} {set start_time [clock seconds]}
                 set time_elapsed [expr {[clock seconds]-$start_time}]
                 if {$time_elapsed >= 10} break
+                # Slow down loop when omem has reached the limit.
+                after 10
+            } else {
+                set start_time 0
             }
         }
         assert {$omem >= 100000 && $time_elapsed < 6}
         $rd1 close
     }
 
+    test {Client output buffer soft limit is enforced when no traffic} {
+        r config set client-output-buffer-limit {pubsub 0 100000 6}
+        set rd1 [redis_deferring_client]
+
+        $rd1 client setname test_client
+        set reply [$rd1 read]
+        assert {$reply eq "OK"}
+
+        $rd1 subscribe foo
+        set reply [$rd1 read]
+        assert {$reply eq "subscribe foo 1"}
+
+        set omem 0
+        set start_time 0
+        set time_elapsed 0
+        while 1 {
+            r publish foo [string repeat "x" 1000]
+            set clients [split [r client list] "\r\n"]
+            set c [lsearch -inline $clients *name=test_client*]
+            if {![regexp {omem=([0-9]+)} $c - omem]} break
```

Comment:
```suggestion
            assert {[regexp {omem=([0-9]+)} $c - omem]}
```
Here the client will not be disconnected.

Reply:
Right. Good fix.
I think this can also be fixed in the first test, the one that makes sure we **don't** disconnect. Right?

Reply:
ohh, i actually merged this one without noticing it (got inside my other merges).
yes, let's apply that on all of them.